src/details/ArborX_Ray.hpp

```diff
@@ -198,6 +199,216 @@ bool intersects(Ray const &ray, Box const &box)
   return intersection(ray, box, tmin, tmax) && (tmax >= 0.f);
 }
 
+// rotate to the x-axis
+KOKKOS_INLINE_FUNCTION Point rotate2D(Point const &point)
+{
+  Point point_star;
+  float r = std::sqrt(point[0] * point[0] + point[1] * point[1]);
+  float costheta = 0.0f;
+  float sintheta = 0.0f;
+  if (r != 0.0f)
+  {
+    costheta = std::fabs(point[0]) / r;
+    sintheta = std::fabs(point[1]) / r;
+  }
+  point_star[0] = point[0] * costheta + point[1] * sintheta;
+  point_star[1] = point[2];
+  point_star[2] = 0.0;
```

Comment:
I don't know how to express it in a more mathematical way. It is really simpler than it looks.

The triangle won't deform because any triangle reaching this step is coplanar with z-axis. 
If the vertices are projected onto the x-y plane they will be on the same line that also passes the origin.

For any point, after rotation, its new x equals its original length.
let's say one point A in x>0, y>0, the new x for A* is r, 
another point B in x<0, y>0, the new x for B* is -r, 
another point C in x>0, y<0, the new x for C* is r.

In this way, I don't need to specify the rotation direction. I just use the above calculation to replace if(x?0,y?0) statements. I agree this might not be the best way to do it.

Reply:
The paper's algorithm rotates the ray so that it becomes `(0, 0, 1)`. I think vertices `A`, `B` and `C` are rotated correspondingly. So I think it's correct to say that the rotated triangle will be co-planar to z-axis. And @wjge calls `rotate2D` on `A` and friends.

Reply:
I am sorry I used the wrong examples in the reply, but I hope you get what I mean. I need to at least at more comments to explain it.